### ISSUE TITLE
Missing /etc/hosts on containers with --network container:name

### ISSUE THREAD

**Issue by fcoelho** [containers/libpod#4626]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

When starting a container with `--network container:name`, dependent containers don't get a `/etc/hosts` file

**Steps to reproduce the issue:**

1. `podman pod create --name=test --share net`
2. `podman run -d --name=netcontainer --pod=test busybox sleep 100000`
3. `podman run --rm --pod=test --network container:netcontainer busybox cat /etc/hosts`

**Describe the results you received:**
```
$ podman run --rm --pod=test --network container:netcontainer busybox cat /etc/hosts
cat: can't open '/etc/hosts': No such file or directory
```

**Describe the results you expected:**

/etc/hosts should be there

**Additional information you deem important (e.g. issue happens only occasionally):**

This happens both with SELinux in "permissive" and "enforcing" mode, no AVCs in audit.log

**Output of `podman version`:**

```
Version:            1.6.2
RemoteAPI Version:  1
Go Version:         go1.13.1
OS/Arch:            linux/amd64

```

**Output of `podman info --debug`:**

```
debug:
  compiler: gc
  git commit: ""
  go version: go1.13.1
  podman version: 1.6.2
host:
  BuildahVersion: 1.11.3
  CgroupVersion: v2
  Conmon:
    package: conmon-2.0.2-1.fc31.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.2, commit: 186a550ba0866ce799d74006dab97969a2107979'
  Distribution:
    distribution: fedora
    version: "31"
  IDMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  MemFree: 2052165632
  MemTotal: 8204685312
  OCIRuntime:
    name: crun
    package: crun-0.10.6-1.fc31.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.10.6
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  SwapFree: 536866816
  SwapTotal: 536866816
  arch: amd64
  cpus: 4
  eventlogger: journald
  hostname: quinto
  kernel: 5.3.13-300.fc31.x86_64
  os: linux
  rootless: true
  slirp4netns:
    Executable: /usr/bin/slirp4netns
    Package: slirp4netns-0.4.0-20.1.dev.gitbbd6f25.fc31.x86_64
    Version: |-
      slirp4netns version 0.4.0-beta.3+dev
      commit: bbd6f25c70d5db2a1cd3bfb0416a8db99a75ed7e
  uptime: 11h 15m 30.16s (Approximately 0.46 days)
registries:
  blocked: null
  insecure: null
  search:
  - docker.io
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - quay.io
store:
  ConfigFile: /home/fcoelho/.config/containers/storage.conf
  ContainerStore:
    number: 3
  GraphDriverName: overlay
  GraphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-0.7-1.fc31.x86_64
      Version: |-
        fusermount3 version: 3.6.2
        fuse-overlayfs: version 0.7
        FUSE library version 3.6.2
        using FUSE kernel interface version 7.29
  GraphRoot: /home/fcoelho/.local/share/containers/storage
  GraphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  ImageStore:
    number: 7
  RunRoot: /run/user/1000
  VolumePath: /home/fcoelho/.local/share/containers/storage/volumes
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-1.6.2-2.fc31.x86_64
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Laptop with Fedora 31 installed a few days ago, the system was fully updated/rebooted before running the commands above

**Comment by mheon** [comment]
Confirmed, repros fine over here, investigating...

**Comment by mheon** [comment]
Fix in #4629

**Comment by fcoelho** [comment]
just an update: tested develop against my original use case (which is a bit more involved than the repro steps above) and it works as expected, thanks for the quick fix!
